src/modules/IscsiClientLib.rb

```diff
@@ -200,37 +200,52 @@ def getFirmwareInfo
       bios_info["stdout"] || ""
     end
 
+    # Takes the output of 'iscsiadm' either in 'node', 'session' or 'fw' mode, i.e.
+    # output where info is seperated by '=', like
+    # iface.transport_name = tcp
+    # iface.hwaddress = 42:52:54:00:8b:6a
+    # iface.bootproto = DHCP
+    #
+    #  @param [String]   stdout (where info is seperated by '=') of iscsiadm command
+    #  @return [Hash]    command output converted to hash
+    #
+    def nodeInfoToMap (stdout)
+      retval = {}
+      return retval if stdout.empty?
+
+      stdout.split("\n").each do |row|
+        key, val = row.split("=")
+
+        key = key.to_s.strip
+        if !key.empty?
+          retval[key] = val.to_s.strip
+        end
+      end
+
+      retval
+    end
+
     # Get iBFT (iSCSI boot firmware table) info
     #
     # @return [Hash] iBFT data converted to a hash (passwords are hidden)
     #
     def getiBFT
       if @ibft == nil
+        @ibft = {}
+
         if !Arch.i386 && !Arch.x86_64
           log.info "Because architecture #{Arch.arch_short} is different from x86, not using iBFT"
-          return {}
+          return @ibft
         end
-        @ibft = {}
-
         ret = SCR.Execute(path(".target.bash_output"),
                           "lsmod |grep -q iscsi_ibft || modprobe iscsi_ibft")
-
         log.info "check and modprobe iscsi_ibft: #{ret}"
 
-        from_bios = getFirmwareInfo.split("\n")
-
-        from_bios.each do |row|
-          key, val = row.split("=")
-
-          key = key.to_s.strip
-          if !key.empty?
-            @ibft[key] = val.to_s.strip
-          end
-        end
+        @ibft = nodeInfoToMap(getFirmwareInfo)
       end
 
       log.info "iBFT: #{hidePassword(@ibft)}"
```

Comment:
with this logic it logs everytime content. I think it make more sense to log it only once when create, so maybe use something like
```
def getiBFT
  return @ibft if @ibft

  ...
```